src/net_processing.cpp

```diff
@@ -3497,45 +3494,41 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
 
         if (pfrom.fSuccessfullyConnected) {
-            // Disconnect peers that send a SENDTXRCNCL message after VERACK.
             LogPrintLevel(BCLog::NET, BCLog::Level::Debug, "sendtxrcncl received after verack from peer=%d; disconnecting\n", pfrom.GetId());
             pfrom.fDisconnect = true;
             return;
         }
 
-        if (!peer->GetTxRelay()) {
-            // Disconnect peers that send a SENDTXRCNCL message even though we indicated we don't
-            // support transaction relay.
+        if (!peer->GetTxRelay() || m_ignore_incoming_txs) {
             LogPrintLevel(BCLog::NET, BCLog::Level::Debug, "sendtxrcncl received from peer=%d to which we indicated no tx relay; disconnecting\n", pfrom.GetId());
             pfrom.fDisconnect = true;
             return;
         }
```

Comment:
It may also happen that we disconnect the peer because `peer->GetTxRelay()` is not set because the peer did not sent us relay=true in their `VERSION` message and we don't provide bloom filter to them. I think this warrants a separate log message, something like:

"sendtxrcncl received from peer=%d that indicated no tx relay; disconnecting\n"